images/cross/Dockerfile

```diff
@@ -28,7 +28,14 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
         rsync \
         runit \
         sudo \
-        zip && \
+        zip
+
+# add nodejs repository using curl
+RUN curl -sL https://deb.nodesource.com/setup_8.x | bash -
+
+# install curl dependency
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -yy -q --no-install-recommends nodejs && \
     DEBIAN_FRONTEND=noninteractive apt-get upgrade -y && \
```

Comment:
when you break this up into multiple steps the intermediate apt-get file will still be in the image layer bloating the container. I think you should try to keep it all in one RUN step:

```
RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
    DEBIAN_FRONTEND=noninteractive apt-get install -yy -q --no-install-recommends \
        awscli \
        ca-certificates \
        curl \
        docker.io \
        git \
        jq \
        lsb-release \
        make \
        rsync \
        runit \
        sudo \
        zip && \
    curl -sL https://deb.nodesource.com/setup_8.x | bash - && \
    DEBIAN_FRONTEND=noninteractive apt-get update && \
    DEBIAN_FRONTEND=noninteractive apt-get install -yy -q --no-install-recommends nodejs && \
    DEBIAN_FRONTEND=noninteractive apt-get upgrade -y && \
    DEBIAN_FRONTEND=noninteractive apt-get autoremove -y && \
    DEBIAN_FRONTEND=noninteractive apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```